pkg/controller/controller_binding.go

```diff
@@ -456,20 +512,9 @@ func (c *controller) reconcileServiceInstanceCredential(binding *v1alpha1.Servic
 					errorReconciliationRetryTimeoutReason,
 					s)
 				c.clearServiceInstanceCredentialCurrentOperation(toUpdate)
-				if _, err := c.updateServiceInstanceCredentialStatus(toUpdate); err != nil {
-					return err
-				}
-
-				// TODO: We need to delete the ServiceInstanceCredential from the
-				// Broker since the Bind request was successful. This needs to be
-				// addressed as part of orphan mitigiation.
-
-				return nil
+				return c.setAndUpdateOrphanMitigation(binding, toUpdate, instance, serviceClass, brokerName, "too much time has elapsed")
 			}
 
```

Comment:
Assuming that we are going to punt for the time being on handling the scenario where there is a failure later that does not prompt an orphan mitigation, can we put a TODO here that says that we need to do something about it in the future?